tests/ship_test.py

```diff
@@ -0,0 +1,235 @@
+        done = False
+        shipClientErrorFile = "%s%d.err" % (shipClientFilePrefix, i)
+        with open(shipClientErrorFile, "r") as errFile:
+            statuses = None
+            try:
+                statuses = json.load(errFile)
+            except json.decoder.JSONDecodeError as er:
+                Utils.errorExit("javascript client output was malformed in %s. Exception: %s" % (shipClientErrorFile, er))
+
+            for status in statuses:
+                statusDesc = status["status"]
+                if statusDesc == "done":
+                    done = True
+                    firstBlockNum = status["first_block_num"]
+                    lastBlockNum = status["last_block_num"]
+                    if maxFirstBN is None:
+                        # initialize both
+                        maxFirstBN = firstBlockNum
+                        minLastBN = lastBlockNum
+                    else:
+                        maxFirstBN = max(maxFirstBN, firstBlockNum)
+                        minLastBN = min(minLastBN, lastBlockNum)
+                    break
+                if statusDesc == "error":
+                    Utils.errorExit("javascript client reporting error see: %s." % (shipClientErrorFile))
+
+        assert done, Print("ERROR: Did not find a \"done\" status for client %d" % (i))
+
+    Print("All clients active from block num: %s to block_num: %s." % (maxFirstBN, minLastBN))
+
+    stderrFile=Utils.getNodeDataDir(shipNodeNum, "stderr.txt")
+    biggestDeltaSeconds = 0.0
+    totalDeltaSeconds = 0.0
+    timeCount = 0
+    with open(stderrFile, 'r') as f:
+        line = f.readline()
+        while line:
+            match = re.search(r'info\s+([0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}.[0-9]{3})\s.+Received\sblock\s+.+\s#([0-9]+)\s@\s([0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}.[0-9]{3})',
+                              line)
+            if match:
+                rcvTimeStr = match.group(1)
+                prodTimeStr = match.group(3)
+                blockNum = int(match.group(2))
+                if blockNum > maxFirstBN:
+                    # ship requests can only affect time after clients started
+                    timeFmt = '%Y-%m-%dT%H:%M:%S.%f'
+                    rcvTime = datetime.strptime(rcvTimeStr, '%Y-%m-%dT%H:%M:%S.%f')
+                    prodTime = datetime.strptime(prodTimeStr, timeFmt)
+                    delta = rcvTime - prodTime
+                    deltaSeconds = delta.total_seconds()
+                    if deltaSeconds > biggestDeltaSeconds:
+                        biggestDeltaSeconds = deltaSeconds
+
+                    totalDeltaSeconds += deltaSeconds
+                    timeCount += 1
+                    assert deltaSeconds < 0.100, Print("ERROR: block_num: %s took more than %.3f seconds to be received." % (blockNum, deltaSeconds))
+
+            line = f.readline()
+
+    avg = totalDeltaSeconds / timeCount if timeCount > 0 else 0.0
```

Comment:
`totalDelta / timeCount` will work as expected.  `avg` will be a `timedelta` object which will print in the format `0:00:00.000000` or `0:00:00,000000` depending on locale so the word 'seconds' can be omitted in the message below.

Reply:
using timeDelta.get_seconds() since 0:00:xx.xxxxxx is really overkill